src/Interpreters/DDLWorker.cpp

```diff
@@ -1010,6 +1099,13 @@ void DDLWorker::cleanupQueue(Int64 current_time_seconds, const ZooKeeperPtr & zo
                 continue;
             }
 
+            /// Skip if not finished
+            if (!isTaskFinished(zookeeper, node_name))
```

Comment:
It may lead to unlimited growth of the queue if there is staled host in a cluster

Reply:
The staled hosts except the one who is not in the cluster, so if user delete it from config file, it will not prevent delete znode .